### PR TITLE
Adding Vuex to book.vue

### PR DESCRIPTION
**Pull request by ahegazy**
Adding Vuex to book.vue
Separating actions,getters,setters from the store/index.js
removing unused plugins

### CODE REVIEW

**Comment by adeen-s** [comment]
Make sure that you are using the latest version of the repo before you make any changes. If you can, please try to make the changes again after you sync your fork with the latest code in the repo.

File: src/store/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Vue from 'vue'
```
Comment (adeen-s) [inline src/store/index.js]: The changes to this file reverts the two latest PRs.

**Review by adeen-s — CHANGES_REQUESTED** [review]

**Comment by ahegazy** [comment]
@adeen-s  I'm using the latest version :D 
I've made some changes .. 
separated the actions,getters,mutations from the main file .. etc ..

File: src/store/index.js (hunk 1 above)
Comment (ahegazy) [reply src/store/index.js]: @adeen-s  I've moved the dotenv to actions.js

**Comment by adeen-s** [comment]
@ahegazy Ok. Please resolve the merge conflicts so that we can merge.

**Comment by ahegazy** [comment]
@adeen-s  done

File: src/store/actions.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+              book = res;
+              commit('grapBook', {
+                bookInfo : book
+              })
+                  });
+            break;
+          default: 
+              this.$router.push('/');
+         }
+
```
Comment (adeen-s) [inline src/store/actions.js]: Consider using only one line to separate parts of code.

File: src/store/actions.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+import axios from 'axios'
+import dotenv from 'dotenv'
+
+dotenv.config()
+
+let $xmltojson = require('../xml2json.min.js')
+let GOODREADS_API_KEY = process.env.GOODREADSKEY // Add your api key here
+
+
+console.log(process.env.GOODREADSKEY || "Fallback");
```
Comment (adeen-s) [inline src/store/actions.js]: If you have verified that the code works, then you could now remove the console.log() line.

File: src/components/book.vue [hunk 4]
```diff
@@ -100,87 +100,20 @@ export default {
...
+          this.$store
+          .dispatch("grapBook", { 
+            src: src,
+            id : id
+           })
+          .then(res => {
+            this.result = this.$store.getters.grapBook;
+          })
+          .catch(err => {
+            console.log("Grap book Failed", err);
```
Comment (adeen-s) [inline src/components/book.vue]: Consider refactoring the code from grapBook to grabBook which is the correct spelling.

**Review by adeen-s — DISMISSED** [review]
Great work @ahegazy Absolutely awesome. I have left some comments but none of them are dealbreakers. I am going to approve this for now. Try to implement those changes in your future commits.

File: src/components/book.vue (hunk 4 above)
Comment (ahegazy) [reply src/components/book.vue]: yeah I felt that something was wrong, thanks :D

**Pull request merged**